Editor/Scripts/Analytics/MeshSyncObserver.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+
+using UnityEditor;
+
+namespace Unity.MeshSync.Editor.Analytics {
+
+    /// <summary>
+    /// Hacky way to restore observers to MeshSync servers in scene
+    /// </summary>
+    [InitializeOnLoad]
+    internal static class MeshSyncObserverStartUp {
+        private const string SESSION_STATE_FLAG = "meshSyncAnalyticsSetupFlag";
+        private const int TIMEOUT_SECONDS = 120;
+        private static DateTime m_check_Start;
+        private static readonly TimeSpan INTERVAL = TimeSpan.FromSeconds(3);
+
+        private static void updateCallback(bool limitChecks) {
+            // Limit checks to once every 3 seconds
+            if ((DateTime.Now - m_check_Start < INTERVAL) && limitChecks) {
+                return;
+            }
+
+            m_check_Start = DateTime.Now;
+
+            var alreadySetup = SessionState.GetBool(SESSION_STATE_FLAG, defaultValue: false);
+
+            // If the setup was already done, we still need to do it again at least once after a domain reload
+            if (alreadySetup) {
+                EditorApplication.update -= checkWithTimeLimits;
+            }
+
+            var array = UnityEngine.Object.FindObjectsOfType<BaseMeshSync>(includeInactive: true);
+            bool success = false;
+
+            foreach (var server in array) {
+                if (server.getNumObservers == 0) {
+                    var observer = new MeshSyncObserver();
+                    server.Subscribe(observer);
+                    success = true;
+                }
+            }
+
+            if (success || EditorApplication.timeSinceStartup > TIMEOUT_SECONDS) {
+                SessionState.SetBool(SESSION_STATE_FLAG, value: true);
+                EditorApplication.update -= checkWithTimeLimits;
+            }
```

Comment:
Since we only check once every 3 seconds, is this relevant anymore?

Reply:
This was removed